src/platform/video/swvideobuffer.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief A software implementation for video buffers.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "swvideobuffer.h"
+#include <cstring>
+
+namespace Thyme
+{
+
+SWVideoBuffer::SWVideoBuffer(Type type) : VideoBuffer(type), m_data(nullptr), m_locked(false) {}
+
+SWVideoBuffer::~SWVideoBuffer()
+{
+    Free();
+}
+
+bool SWVideoBuffer::Allocate(unsigned width, unsigned height)
+{
+    Free();
+    m_width = width;
+    m_height = height;
+    m_textureWidth = width;
+    m_textureHeight = height;
+
+    if (m_format != VideoBuffer::TYPE_X8R8G8B8 && m_format != VideoBuffer::TYPE_R8G8B8) {
```

Comment:
Perhaps move this check to the top of the function, so that nothing is changed if it will fail anyway.